src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -60,6 +61,8 @@ def __init__(
             Normal,
         ],
         initrv: typing.Optional[Normal] = None,
+        diffusion=None,
+        re_predict_with_calibrated_diffusion=True,
```

Comment:
this flag allows switching between the dynamic and pseudo-dynamic diffusions that we talked about before

Reply:
Should we really ad this as an option, even though it was mostly just for us to experiment with stuff? If we want to fully support that, we should probably rather implement another diffusion model?

Reply:
Do you have an idea how to put this into a diffusion model? 

I do not really; in fact, I think this current way is the cleanest way of handling this re-prediction,  because only this way can we use diffusions that are not informed by any ODE solver/ transition behaviour. Do you see what I mean here? 

On a side note, this is also kind of how I think about these things mathematically: re-prediction with a calibrated diffusion is always an option (even though with a constant diffusion this does not really make a lot of sense). What do you think?

Reply:
I think I don't agree with "re-prediction with a calibrated diffusion is always an option". Depending on the model you consider, there should be a right way to do things: The dynamic diffusion needs to be considerered in a prediction, but the constant diffusion can be disregarded if we calibrate at the end. 

I'm also not 100% sure what the "pseudo-dynamic diffusion" means exactly, which is my main reason to why I'm careful to add a support for it. How would we write it down?

Reply:
I added a (IMO) rather thorough docstring for `step()`. Does that clear things up now? If not, let's talk in person tomorrow.